### PR TITLE
Fix for api changes in ballerina-lang

### PR DESCRIPTION
**Pull request by riyafa**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java [hunk 1]
```diff
@@ -64,25 +67,50 @@ public boolean onMessage(EmailEvent emailEvent) {
...
+                invokeMethodSync(service.getValue(), ON_MESSAGE, ON_MESSAGE_METADATA, email, true);
             }
         } else {
             log.error("Runtime should not be null.");
         }
         return true;
     }
 
+    private void invokeMethodSync(ObjectValue object, String methodName, StrandMetadata metadata, Object... args) {
+        Semaphore semaphore = new Semaphore(0);
```
Comment (Maninda) [inline email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java:79]: Declaring a new Semaphore in each method call may not limit the access. May be we can declare it only once.

File: email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java [hunk 2]
```diff
@@ -64,25 +67,50 @@ public boolean onMessage(EmailEvent emailEvent) {
...
+            }
+
+            @Override
+            public void notifyFailure(ErrorValue error) {
+                errorValue[0] = error;
+                semaphore.release();
+            }
+        }, args);
+        try {
+            semaphore.acquire();
```
Comment (Maninda) [inline email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java:94]: `invokeMethodAsync` may be required to be called after this line as the Semaphore was not yet acquired at that time.

File: email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java [hunk 3]
```diff
@@ -64,25 +67,50 @@ public boolean onMessage(EmailEvent emailEvent) {
...
+        } catch (InterruptedException e) {
+            // Ignore
```
Comment (Maninda) [inline email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java:96]: Can we handle this case and throw the errors?

File: email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java [hunk 4]
```diff
@@ -64,25 +67,50 @@ public boolean onMessage(EmailEvent emailEvent) {
...
-                runtime.invokeMethodSync(service.getValue(), ON_MESSAGE, null, ON_MESSAGE_METADATA, email, true);
+                invokeMethodSync(service.getValue(), ON_MESSAGE, ON_MESSAGE_METADATA, email, true);
             }
         } else {
             log.error("Runtime should not be null.");
         }
         return true;
     }
 
+    private void invokeMethodSync(ObjectValue object, String methodName, StrandMetadata metadata, Object... args) {
```
Comment (Maninda) [inline email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java:78]: This method may require to throw an error as an error is thrown inside this method.

File: email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java (hunk 1 above)
Comment (riyafa) [reply email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java:79]: This method is to replace the removed function here: https://github.com/ballerina-platform/ballerina-lang/pull/25744/files#diff-a3b2c474ab6a7c3e36374099c3bf5e9eL208-L233

**Comment by riyafa** [comment]
@Maninda if you think async method is more appropriate in your connector it is highly recommended to use it. If the connector needs to block, I have ported the sync method here: https://github.com/ballerina-platform/ballerina-lang/pull/25744/files#diff-a3b2c474ab6a7c3e36374099c3bf5e9eL208-L233 
 We are no longer supporting the sync method in runtime

File: email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java (hunk 2 above)
Comment (riyafa) [reply email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java:94]: The semaphore is released inside a callback. This is ported from the removed function from the runtime api. If you think async method is more appropriate, its use is recommended.

**Comment by riyafa** [comment]
Async methods are now used in the PR #11

**Pull request closed without merging**